### ISSUE TITLE
We should be able to sub-select features based on strongest correlation with outcome or some other criteria (e.g., `select_k_best` in `sklearn`)

### ISSUE THREAD

**Issue by mmcdermott** [mmcdermott/MEDS_Tabular_AutoML#46]

**Comment by mmcdermott** [comment]
@teyaberg I feel like this was added, right? Or am I totally wrong?

**Comment by teyaberg** [comment]
Yes, we have approximate correlation feature selection implemented through [this function](https://github.com/mmcdermott/MEDS_Tabular_AutoML/blob/930191399f726c6ef3f3a02161e5e4930d985d18/src/MEDS_tabular_automl/tabular_dataset.py#L217) and this can be used to define the allowed codes [here](https://github.com/mmcdermott/MEDS_Tabular_AutoML/blob/930191399f726c6ef3f3a02161e5e4930d985d18/src/MEDS_tabular_automl/tabular_dataset.py#L179).
